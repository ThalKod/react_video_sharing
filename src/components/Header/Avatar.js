import React from "react";

export default () => {
  const style = {
    avatar:{
      backgroundColor: "#a2aec1",
      width: 50,
      height: 50,
      borderRadius: 50/2,
      textAlign: "center",

    },
    text: {
      color: "white",
      fontSize: 15,
      marginTop: 5,
    }
  };
  return (
        <div style={style.avatar} className="avatar pull-left">
          <div style={style.text}>
            TM
          </div>
        </div>
  )
};
