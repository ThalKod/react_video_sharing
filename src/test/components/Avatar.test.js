import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import Avatar from "components/Header/Avatar";

describe("Avatar", () => {
  let wrapper;
  beforeEach(() => {
    const username = "John Doe";
    wrapper = shallow(<Avatar username={username}/>);
  });

  it("should render the Avatar component correctly", () => {
      expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should render the correct initials", () => {
    expect(
        wrapper.containsMatchingElement(<div>JD</div>)
    ).toBe(true);
  });
});
