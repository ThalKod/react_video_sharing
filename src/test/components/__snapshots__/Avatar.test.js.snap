// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar should render the Avatar component correctly 1`] = `
<div
  className="avatar pull-left"
  style={
    Object {
      "backgroundColor": "#a2aec1",
      "borderRadius": 25,
      "height": 50,
      "marginRight": 15,
      "marginTop": 10,
      "textAlign": "center",
      "width": 50,
    }
  }
>
  <div
    style={
      Object {
        "color": "white",
        "fontSize": 15,
        "marginTop": 5,
      }
    }
  >
    JD
  </div>
</div>
`;
